src/main/java/org/opentripplanner/middleware/triptracker/TravelerLocator.java

```diff
@@ -130,16 +129,12 @@ private static TripInstruction getBackOnTrack(
     @Nullable
     public static TripInstruction alignTravelerToTrip(
         TravelerPosition travelerPosition,
-        boolean isStartOfTrip,
-        TripStatus tripStatus
+        boolean isStartOfTrip
     ) {
         Locale locale = travelerPosition.locale;
 
         if (isApproachingEndOfLeg(travelerPosition)) {
-            if (isBusLeg(travelerPosition.nextLeg) && isWithinOperationalNotifyWindow(travelerPosition)) {
-                BusOperatorActions
-                    .getDefault()
-                    .handleSendNotificationAction(tripStatus, travelerPosition);
+            if (sendBusNotification(travelerPosition, isStartOfTrip)) {
```

Comment:
You might have misinterpreted `isStartOfTrip` when using with `sendBusNotification`, and that is what is causing the failing trip tracking E2E test. I think `true` means that this is the first tracking location that is sent (the user just activated live tracking). I think you meant to pass to `sendBusNotification` a boolean indicating whether a transit leg is the first leg of the trip.

Reply:
@binh-dam-ibigroup I have updated, I hope it meets expectations.